### ISSUE TITLE
get ctc_greedy_decoder results error C-api ?

### ISSUE THREAD

**Issue by kvcuong** [tensorflow/tensorflow#24472]
**System information**
- Have I written custom code 
- OS Platform and Distribution (Linux Ubuntu 16.04):
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.12
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source): 
- CUDA/cuDNN version: not available
- GPU model and memory: not available

Python code:
self.seqLen = tf.placeholder(tf.int32, [None])
decoder = tf.nn.ctc_greedy_decoder(inputs=ctcIn3dTBC, sequence_length=self.seqLen)

Python result:
`([SparseTensorValue(indices=array([[ 0,  0],
       [ 0,  1],
       [ 0,  2],
       [ 0,  3],
       [ 1,  0],
       [ 1,  1],
       [ 1,  2],
       [ 1,  3],
       [ 2,  0],
       [ 2,  1],
       [ 2,  2],
       [ 2,  3],
       [ 3,  0],
       [ 3,  1],
       [ 3,  2],
       [ 3,  3],
       [ 4,  0],
       [ 4,  1],
       [ 4,  2],
       [ 4,  3],
       [ 5,  0],
       [ 5,  1],
       [ 5,  2],
       [ 5,  3],
       [ 6,  0],
       [ 6,  1],
       [ 6,  2],
       [ 6,  3],
       [ 7,  0],
       [ 7,  1],
       [ 7,  2],
       [ 7,  3],
       [ 8,  0],
       [ 8,  1],
       [ 8,  2],
       [ 8,  3],
       [ 9,  0],
       [ 9,  1],
       [ 9,  2],
       [ 9,  3],
       [10,  0],
       [10,  1],
       [10,  2],
       [10,  3],
       [11,  0],
       [11,  1],
       [11,  2],
       [11,  3],
       [12,  0],
       [12,  1],
       [12,  2],
       [12,  3],
       [13,  0],
       [13,  1],
       [13,  2],
       [13,  3],
       [14,  0],
       [14,  1],
       [14,  2],
       [14,  3],
       [15,  0],
       [15,  1],
       [15,  2],
       [15,  3],
       [16,  0],
       [16,  1],
       [16,  2],
       [16,  3],
       [17,  0],
       [17,  1],
       [17,  2],
       [17,  3],
       [18,  0],
       [18,  1],
       [18,  2],
       [18,  3],
       [19,  0],
       [19,  1],
       [19,  2],
       [19,  3],
       [20,  0],
       [20,  1],
       [20,  2],
       [20,  3],
       [21,  0],
       [21,  1],
       [21,  2],
       [21,  3],
       [22,  0],
       [22,  1],
       [22,  2],
       [22,  3],
       [23,  0],
       [23,  1],
       [23,  2],
       [23,  3],
       [24,  0],
       [24,  1],
       [24,  2],
       [24,  3],
       [25,  0],
       [25,  1],
       [25,  2],
       [25,  3],
       [26,  0],
       [26,  1],
       [26,  2],
       [26,  3],
       [27,  0],
       [27,  1],
       [27,  2],
       [27,  3],
       [28,  0],
       [28,  1],
       [28,  2],
       [28,  3],
       [29,  0],
       [29,  1],
       [29,  2],
       [29,  3],
       [30,  0],
       [30,  1],
       [30,  2],
       [30,  3],
       [31,  0],
       [31,  1],
       [31,  2],
       [31,  3],
       [32,  0],
       [32,  1],
       [32,  2],
       [32,  3],
       [33,  0],
       [33,  1],
       [33,  2],
       [33,  3],
       [34,  0],
       [34,  1],
       [34,  2],
       [34,  3],
       [35,  0],
       [35,  1],
       [35,  2],
       [35,  3],
       [36,  0],
       [36,  1],
       [36,  2],
       [36,  3],
       [37,  0],
       [37,  1],
       [37,  2],
       [37,  3],
       [38,  0],
       [38,  1],
       [38,  2],
       [38,  3],
       [39,  0],
       [39,  1],
       [39,  2],
       [39,  3],
       [40,  0],
       [40,  1],
       [40,  2],
       [40,  3],
       [41,  0],
       [41,  1],
       [41,  2],
       [41,  3],
       [42,  0],
       [42,  1],
       [42,  2],
       [42,  3],
       [43,  0],
       [43,  1],
       [43,  2],
       [43,  3],
       [44,  0],
       [44,  1],
       [44,  2],
       [44,  3],
       [45,  0],
       [45,  1],
       [45,  2],
       [45,  3],
       [46,  0],
       [46,  1],
       [46,  2],
       [46,  3],
       [47,  0],
       [47,  1],
       [47,  2],
       [47,  3],
       [48,  0],
       [48,  1],
       [48,  2],
       [48,  3],
       [49,  0],
       [49,  1],
       [49,  2],
       [49,  3]], dtype=int64), values=array([53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53,
       59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59,
       64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64,
       57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57,
       53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53,
       59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59,
       64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64,
       57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57,
       53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53,
       59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59,
       64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64,
       57, 53, 59, 64, 57, 53, 59, 64, 57, 53, 59, 64, 57], dtype=int64), dense_shape=array([50,  4], dtype=int64))], array([[-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455],
       [-297.2455]], dtype=float32))`

Graph-def:
`483: CTCGreedyDecoder type: CTCGreedyDecoder device:  number inputs: 2 number outputs: 4
Number inputs: 2
0 type : TF_FLOAT
1 type : TF_INT32
Number outputs: 4
0 type : TF_INT64
 dims: 2 [-1,2]
1 type : TF_INT64
 dims: 1 [-1]
2 type : TF_INT64
 dims: 1 [2]
3 type : TF_FLOAT
 dims: 2 [50,1]`

C-API result:

raw ouput:
0 0
1 0
2 0
3 0
4 0
5 0
6 0
7 0
8 0
9 0
10 0
11 0
12 0
13 0
14 0
15 0
16 0
17 0
18 0
19 0
20 0
21 0
22 0
23 0
24 0
25 0
26 0
27 0
28 0
29 0
30 0
31 0
32 0
33 0
34 0
35 0
36 0
37 0
38 0
39 0
40 0
41 0
42 0
43 0
44 0
45 0
46 0
47 0
48 0
49 0
-1759315024 1332898372
-1746890864 -1804888032
-1759345360 -1759345360
-1759317136 -1759317136
-1759345360 -1759345360
-1759343376 -1759343376
-572662307 -572662307
-572662307 -572662307
-572662307 2104715895
544096522 544498024
1702258025 1768778100
1881173615 1919251557
1696623983 1735289188
1953722216 779711087
1700929652 1701078371
1126195488 1701999975
1684365933 1953722184
1952540788 1886724211
1919902496 1651336480
1948280686 1852793632
1769234802 544370502
744844400 1886330995
1684370293 1920230770
1685091616 1918988320
544436837 1931502962
543236199 1768910955
1918988298 980644453
1869835877 1767994478
1701344288 1768186981
1701601889 1952804193
1931505524 1953068832
1634732645 1936876916
1835343981 543649385
544433524 1752440935
1634887521 1768778100
1629515375 1835562089
1970107747 540701554
1919906670 1852399988
543516788 1852400740
543517794 1634497901
544175136 1769414757
1879729512 1919251557
1696623983 1735289188
1936028769 1948280686
1919377764 1835627632
544108393 1752459634
1701601889 1310734949
1948280431 1818386804
544502645 543236200
544106784 1347690528
1768186981 1734960750
544108393 1864900719
1936024681 1767859564
1601009006 540701540
1864397413 1935962721
1752637551 543516788
1852400740 1936026722
1986618400 1752369710
540697705 1768319348
1752375398 544370534
1701867296 170815087
1684627301 544761188
544434536 1852383333
1651328288 1632397166
1768320623 1852795252
537554804 1667592816
667177 1768319348
1752375398 1081240139
-1804996768 -1804848800
1684627301 544761188
544434536 1852383333
1651328288 1632397166
1768320623 1852795252
537554804 1667592816
667177 74672654
544096522 544498024
1886330995 1769234810
1835102817 1852383347
1713392962 1684365933
1967988782 1919945317
1646290021 1718513475
1431327845 1852400740
1886330996 1702043764
1752440944 1667592818
1684301154 1818386804
1734960750 778989417
1635280160 1752440876
1936269424 1869881444
1819042164 1952804193
1952540788 1634692128
1836020326 1801676136
1700929652 1948279072
543649385 175335712
1684370549 1835102817
1092631155 544370547
1768843617 1763730792
1696623713 1735289188
1881171308 1919251557
1702065440 1700949349
1869351527 1937055859
543516788 1953525536
1869182049 1769107303
1836018954 541139002
1663070831 1852403305
1852383333 1835343980
543649385 1869422693
1869881441 544106784
1852400740 1886743407
543649385 1296122945
2053729641 1818304622
778922100 1953063791
1948270880 1868767346
1735289198 1768843552
1651336480 1948280686
1818588704 544433513
1763730803 1684301154
1802465132 1852404597
1145118821 1835627632
544108393 1752459634
1701601889 1310734949
1948280431 1818386804
544502645 543236200
544106784 1347690528
1768186981 1734960750
544108393 1864900719
1936024681 1767859564
1601009006 540701540
1864397413 1935962721
1752637551 543516788
1852400740 1936026722
1986618400 1752369710
540697705 1768319348
1752375398 544370534
1701867296 170815087
1684627301 544761188
544434536 1852383333
1651328288 1632397166
1768320623 1852795252
537554804 1667592816
667177 0
0 175335936
544096522 544498024
1886330995 1769234810
1835102817 1852383347
1713392962 1684365933
1967988782 1919945317
1646290021 1718513475
1431327845 1852400740
1886330996 1702043764
1752440944 1667592818

I'am blocked to correctly get  result of ctc_greedy_decoder in c-API, please help?
Thanks.

**Comment by kvcuong** [comment]
I found the problem. I forgot to normalize the input batch data.
